server/controllers/bookController.js

```diff
@@ -7,43 +7,75 @@ const { Op } = sequelize;
 
 class BookController {
   static async addBook(req, res) {
-    const { title, description, amount, authorID, status, year, categoryID, isbn } = req.body;
-    const book = await models.Books.create({
-      title,
-      description,
-      categoryID,
-      amount,
-      authorID,
-      status,
-      year,
-      isbn
-    });
-    return Utils.successStatus(res, 201, 'Book added successfully', book);
-  }
+    const { title, description, tags, amount, authorID, status, year, categoryID, isbn } = req.body;
+      const book = await models.Books.create({
+        title,
+        description,
+        tags,
+        categoryID,
+        amount,
+        authorID,
+        status,
+        year,
+        isbn
+      });
+      return Utils.successStatus(res, 201, 'Book added successfully', book);
+    }
 
   static async getAllBooks(req, res) {
-    if (req.query.page) {
+    if (req.query) {
       let { page, limit } = req.query;
       page = parseInt(page, 10);
       page = page > 0 ? page : 1;
       limit = parseInt(limit, 10);
       limit = limit > 0 ? limit : 50;
       const offset = (page - 1) * limit;
-      const books = await models.Books.findAndCountAll({
-        limit,
-        offset,
-        $sort: { id: 1 }
-      });
-      const pages = Math.ceil(books.count / limit);
-      const data = books.rows;
-      return res.status(200).json({
-        status: res.status,
-        message: 'success',
-        pages,
-        data
+
+      if (req.query.searchBook) {         
+        const searchResult = await models.Books.findAll({
+          limit,
+          offset,
+          $sort: { id: 1 },
+          include: [{
+              model: models.Authors,
+              as: 'Author',
+              where: { 
+                [Op.or]: [
+                  {firstName: {[Op.like]: `%${req.query.searchBook}%`}},
+                  {middleName: {[Op.like]: `%${req.query.searchBook}%`}},
+                  {lastName: {[Op.like]: `%${req.query.searchBook}%`}},
+                  {'$Books.title$': {[Op.like]: `%${req.query.searchBook}%`}},
+                  {'$Books.description$': {[Op.like]: `%${req.query.searchBook}%`}},
+                  {'$Books.tags$': {[Op.like]: `%${req.query.searchBook}%`}},
+                ]
+               }
+          }]
       });
+        if (!searchResult[0]) {
+        return Utils.errorStatus(res, 404, 'No record found')  
+      }
+          return Utils.successStatus(res, 200, 'Books fetched successfully', searchResult) 
```

Comment:
The pagination is returned here if exists

Reply:
So I guess this should come after line 75

Reply:
Are you referring to line 57?